lib/IRGen/GenDecl.cpp

```diff
@@ -3595,8 +3609,10 @@ llvm::GlobalValue *IRGenModule::defineTypeMetadata(CanType concreteType,
   if (nominal)
     addRuntimeResolvableType(nominal);
 
-  // Don't define the alias for foreign type metadata, since it's not ABI.
-  if (nominal && requiresForeignTypeMetadata(nominal))
+  // Don't define the alias for foreign type metadata or prespecialized generic
+  // metadata, since neither is ABI.
+  if ((nominal && requiresForeignTypeMetadata(nominal)) ||
+      concreteType->is<BoundGenericType>())
```

Comment:
Another suspicious use of BoundGenericType.

Reply:
Yeah, that limitation is everywhere at the moment.  Would it be reasonable to start out only prespecializing these cases and expand from here?